### PR TITLE
Onomy enhancements

### PR DESCRIPTION
**Pull request by DavidNix**
## Onomy
* Adds gbt companion process required by validators
* Closes https://github.com/strangelove-ventures/heighliner/issues/68

### CODE REVIEW

**Comment by agouin** [comment]
`--target` is a flag specific to cargo. The required flag also means that multi-line `build-target` will not work for rust. I posted my reasoning on not touching rust for now [here](https://github.com/strangelove-ventures/heighliner/pull/78#issuecomment-1402767318).

If we want to handle this now, we should do something like export a `CARGO_TARGET` env var which can be used by the specific chain `build-target` cargo commands

File: chains.yaml [hunk 1]
```diff
@@ -545,10 +545,23 @@
   github-organization: onomyprotocol
   github-repo: onomy
   dockerfile: cosmos
+  # TODO: Heighliner will not produce working builds until https://github.com/onomyprotocol/onomy/pull/129 merged.
+  # If PR not merged for your git-ref, then update and push to our fork, https://github.com/strangelove-ventures/onomy, with Onomy upstream.
+  # And manually run heighliner command with `-o strangelove-ventures --git-ref main --tag <semver>`.
   build-target: make install
```
Comment (DavidNix) [inline chains.yaml:551]: @agouin Decided to take this approach vs. a long build target. 

As of this writing, https://github.com/onomyprotocol/onomy/pull/129 has no activity from the Onomy team. I've reached out to them in our dedicated slack channel.

**Review by danbryan — APPROVED** [review]

File: chains.yaml [hunk 2]
```diff
@@ -545,10 +545,23 @@
...
   binaries:
     - /go/bin/onomyd
 
+# Onomy gbt: Not a chain, companion process required by validator
+- name: onomy-gbt
+  github-organization: onomyprotocol
+  github-repo: arc
+  dockerfile: rust
```
Comment (agouin) [inline chains.yaml:559]: This will work but `rust` is now deprecated. `cargo` replaces it.

Comment (DavidNix) [reply chains.yaml:559]: Thanks, I missed that deprecation. I'll make a quick update.

**Pull request merged**